main/src/library/MutSet.flix

```diff
@@ -212,7 +206,7 @@ namespace MutSet {
     ///
     /// That is, the result is of the form: `f(...f(f(i, x1), x2)..., xn)`.
     ///
-    pub def foldLeft(f: (b, a) -> b & ef, i: b, s: MutSet[a]): b & Impure =
+    pub def foldLeft(f: (b, a) -> b & ef, i: b, s: MutSet[a, r]): b & reg(r) and ef =
```

Comment:
I think I prefer to always write this in the order `ef and reg(r)`, i.e. since `ef` occurs before `r` in the arguments let us also do it in the types.